helix-core/src/main/java/org/apache/helix/controller/rebalancer/topology/Topology.java

```diff
@@ -289,6 +310,29 @@ private boolean isInstanceEnabled(ClusterConfig clusterConfig, String instanceNa
     return instanceTopologyMap;
   }
 
+  /**
+   * This function returns a LinkedHashMap<String, String> object representing
+   * the topology path for an instance.
+   * LinkedHashMap is used here since the order of the path needs to be preserved
+   * when creating the topology tree.
+   *
+   * @param clusterConfig         clusterConfig of the given cluster.
+   * @param instanceName          name of the instance.
+   * @param instanceConfig        instanceConfig to be checked.
+   * @param earlyQuitForFaultZone Set to true if we only need the path till faultZone.
+   *
+   * @return an LinkedHashMap object representing the topology path for the input instance.
+   */
+  public static LinkedHashMap<String, String> computeInstanceTopologyMap(
+      ClusterConfig clusterConfig, String instanceName, InstanceConfig instanceConfig,
+      boolean earlyQuitForFaultZone) throws IllegalArgumentException {
```

Comment:
I suggest not throwing `IllegalArgumentException` in the method signature because: an unchecked exception does not need to declare in the method signature. `throws` keyword is used for handling checked exception. There are some good articles about this, eg: https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#throwstag

I suggest put this info in javadoc like this 
```
@throws IllegalArugmentException  when it is thrown
```

Reply:
TFTR. Updated.